js/grid.js

```diff
@@ -215,6 +216,16 @@
     },
 
     /**
+     * Test a tile at a given position for being pinned or history
+     *
+     * @param {Number} aPos Position in sites array
+     */
+    _isHistoricalTile(aPos) {
+      let site = this.sites[aPos];
+      return site && (site.isPinned() || site.link && site.link.type === "history");
+    },
+
+    /**
      * Make sure the correct number of rows and columns are visible
      */
     _resizeGrid() {
```

Comment:
This all seems like it's just replicating the behavior that should be handled by flexbox? We should see if we can get rid of this using CSS.